modules/fedimint-mint-client/src/lib.rs

```diff
@@ -80,10 +81,84 @@ pub const LOG_TARGET: &str = "client::module::mint";
 /// An encapsulation of [`FederationId`] and e-cash notes in the form of
 /// [`TieredMulti<SpendableNote>`] for the purpose of spending e-cash
 /// out-of-band. Also used for validating and reissuing such out-of-band notes.
+///
+/// ## Invariants
```

Comment:
Would be nice if we had it in code. 

Make a `fs check_invariants`, `impl Decodable` manually to use it, etc.?